TestApp/src/main/java/com/tekchand/testapp/ui/main/SessionFrag.java

```diff
@@ -0,0 +1,193 @@
+
+import java.util.ArrayList;
+
+import static com.tekchand.testapp.ui.main.LogIn.EMAIL;
+import static com.tekchand.testapp.ui.main.LogIn.NAME;
+import static com.tekchand.testapp.ui.main.LogIn.sharedPreferences;
+
+
+/**
+ * A simple {@link Fragment} subclass.
+ * Activities that contain this fragment must implement the
+ * {@link OnFragmentInteractionListener} interface
+ * to handle interaction events.
+ * Use the {@link SessionFrag#newInstance} factory method to
+ * create an instance of this fragment.
+ */
+public class SessionFrag extends Fragment {
+    // TODO: Rename parameter arguments, choose names that match
+    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+
+    private static final String ARG_PARAM1 = "param1";
+    private static final String ARG_PARAM2 = "param2";
+
+
+    // TODO: Rename and change types of parameters
+    private String mParam1;
+    private String mParam2;
+
+    private OnFragmentInteractionListener mListener;
+
+    public SessionFrag() {
+        // Required empty public constructor
+    }
+
+    /**
+     * Use this factory method to create a new instance of
+     * this fragment using the provided parameters.
+     *
+     * @param param1 Parameter 1.
+     * @param param2 Parameter 2.
+     * @return A new instance of fragment SessionFrag.
+     */
+    // TODO: Rename and change types and number of parameters
+    public static SessionFrag newInstance(String param1, String param2) {
+        SessionFrag fragment = new SessionFrag();
+        Bundle args = new Bundle();
+        args.putString(ARG_PARAM1, param1);
+        args.putString(ARG_PARAM2, param2);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (getArguments() != null) {
+            mParam1 = getArguments().getString(ARG_PARAM1);
+            mParam2 = getArguments().getString(ARG_PARAM2);
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+
+        View root = inflater.inflate(R.layout.fragment_session, container, false);
+        return root;
+    }
+
+
+    TextView userName;
+    TextView email;
+    Button logOutBtn;
+
+
+    @Override
+    public void onViewCreated(@NonNull View root, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(root, savedInstanceState);
+        logOutBtn = root.findViewById(R.id.logoutbutton);
+        userName = root.findViewById(R.id.userNameTextView2);
+        email = root.findViewById(R.id.emailTextView2);
+    }
+
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+
+        userName.setText("User Name: " + sharedPreferences.getString(NAME, null));
```

Comment:
Create a SharedPreference manager class. Activity or Fragment should not interact with SharedPreferences directly.